rust/metaplex/program/src/utils.rs

```diff
@@ -176,6 +177,39 @@ pub fn assert_authority_correct(
 
     Ok(())
 }
+
+pub fn assert_auction_is_ended_or_valid_instant_sale(
+    auction_info: &AccountInfo,
+    auction_extended_info: Option<&AccountInfo>,
+    bidder_metadata_info: &AccountInfo,
+    win_index: Option<usize>,
+) -> ProgramResult {
+    if AuctionData::get_state(auction_info)? == AuctionState::Ended {
+        return Ok(());
+    }
+
+    let instant_sale_price = auction_extended_info
+        .and_then(|info| AuctionDataExtended::get_instant_sale_price(&info.data.borrow()));
+
+    match instant_sale_price {
+        Some(instant_sale_price) => {
+            let winner_bid_price = if let Some(win_index) = win_index {
+                AuctionData::get_winner_bid_amount_at(auction_info, win_index).unwrap()
+            } else {
+                // Possible case in an open auction
+                BidderMetadata::from_account_info(bidder_metadata_info)?.last_bid
```

Comment:
Nice, love to see the edge case handling.

Reply:
Sorry, which cases do u mean? if understand correctly, all we need to know `winner_bid_price < instant_sale_price`

Reply:
Oh I wasn't saying you were missing anything. It was a compliment.